src/graphql/root/mutation/onchain-usd-payment-send.ts

```diff
@@ -0,0 +1,86 @@
+import { GT } from "@graphql/index"
+import Memo from "@graphql/types/scalar/memo"
+import { mapAndParseErrorForGqlResponse } from "@graphql/error-map"
+import WalletId from "@graphql/types/scalar/wallet-id"
+import CentAmount from "@graphql/types/scalar/cent-amount"
+import OnChainAddress from "@graphql/types/scalar/on-chain-address"
+import PaymentSendPayload from "@graphql/types/payload/payment-send"
+import TargetConfirmations from "@graphql/types/scalar/target-confirmations"
+import { validateIsUsdWalletForMutation } from "@graphql/helpers"
+import { Wallets } from "@app"
+
+const OnChainUsdPaymentSendInput = GT.Input({
+  name: "OnChainUsdPaymentSendInput",
+  fields: () => ({
+    walletId: { type: GT.NonNull(WalletId) },
+    address: { type: GT.NonNull(OnChainAddress) },
+    amount: { type: GT.NonNull(CentAmount) },
+    memo: { type: Memo },
+    targetConfirmations: { type: TargetConfirmations, defaultValue: 1 },
+  }),
+})
+
+const OnChainUsdPaymentSendMutation = GT.Field<
+  {
+    input: {
+      walletId: WalletId | InputValidationError
+      address: OnChainAddress | InputValidationError
+      amount: number
+      memo: Memo | InputValidationError | null
+      targetConfirmations: TargetConfirmations | InputValidationError
+    }
+  },
+  null,
+  GraphQLContextAuth
+>({
+  extensions: {
+    complexity: 120,
+  },
+  type: GT.NonNull(PaymentSendPayload),
+  args: {
+    input: { type: GT.NonNull(OnChainUsdPaymentSendInput) },
+  },
+  resolve: async (_, args, { domainAccount }) => {
+    const { walletId, address, amount, memo, targetConfirmations } = args.input
+
+    if (walletId instanceof Error) {
+      return { errors: [{ message: walletId.message }] }
+    }
+
+    if (address instanceof Error) {
+      return { errors: [{ message: address.message }] }
+    }
+
+    if (memo instanceof Error) {
+      return { errors: [{ message: memo.message }] }
+    }
+
+    if (targetConfirmations instanceof Error) {
+      return { errors: [{ message: targetConfirmations.message }] }
+    }
+
+    const usdWalletValidated = await validateIsUsdWalletForMutation(walletId)
```

Comment:
this validation should be part of app layer, also `validateIsUsdWalletForMutation` is using wallet repository directly so that method should be removed and be included in use cases

Reply:
Oh yea good point. These are used in lots of mutation/queries here. Maybe we can change all of these in a follow-up PR to keep the diff in this one clear?